nifi-nar-bundles/nifi-amqp-bundle/nifi-amqp-processors/src/main/java/org/apache/nifi/amqp/processors/PublishAMQP.java

```diff
@@ -195,29 +193,85 @@ public void process(final InputStream in) throws IOException {
-                String amqpPropName = attributeEntry.getKey().split("\\" + AMQPUtils.AMQP_PROP_DELIMITER)[1];
+                String amqpPropName = attributeEntry.getKey();
                 String amqpPropValue = attributeEntry.getValue();
-                try {
-                    if (amqpPropertyNames.contains(AMQPUtils.AMQP_PROP_PREFIX + amqpPropName)) {
-                        Method m = builder.getClass().getDeclaredMethod(amqpPropName, String.class);
-                        m.invoke(builder, amqpPropValue);
-                    } else {
-                        getLogger().warn("Unrecogninsed AMQP property '" + amqpPropName + "', will ignore.");
+
+                AMQPUtils.PropertyNames propertyNames = AMQPUtils.PropertyNames.fromValue(amqpPropName);
+
+                if (propertyNames != null) {
+                    try {
+                        switch (propertyNames){
+                            case CONTENT_TYPE:
+                                builder.contentType(amqpPropValue);
+                                break;
+                            case CONTENT_ENCODING:
+                                builder.contentEncoding(amqpPropValue);
+                                break;
+                            case HEADERS:
+                                String[] s = amqpPropValue.split("\\" + AMQPUtils.AMQP_PROP_DELIMITER);
+                                Map<String, Object> headers = new HashMap<>();
+
+                                for (int i = 0; i < s.length ; i += 2){
+                                    if (i + 2 <= s.length){
+                                        headers.put(s[i], s[i + 1]);
+                                    }
+                                }
+
+                                builder.headers(headers);
+                                break;
+                            case DELIVERY_MODE:
+                                builder.deliveryMode(Integer.parseInt(amqpPropValue));
+                                break;
+                            case PRIORITY:
+                                builder.priority(Integer.parseInt(amqpPropValue));
+                                break;
+                            case CORRELATION_ID:
+                                builder.correlationId(amqpPropValue);
+                                break;
+                            case REPLY_TO:
+                                builder.replyTo(amqpPropValue);
+                                break;
+                            case EXPIRATION:
+                                builder.expiration(amqpPropValue);
+                                break;
+                            case MESSAGE_ID:
+                                builder.messageId(amqpPropValue);
+                                break;
+                            case TIMESTAMP:
+                                builder.timestamp(new SimpleDateFormat("yyyy/MM/dd HH:mm:ss").parse(amqpPropValue));
```

Comment:
I see that the assumption here is that String representation of Date will always come in "yyyy/MM/dd HH:mm:ss". There are several issues here:
1. Similar to DELIVERY_MODE and PRIORITY, the value could be null resulting in NPE
2. Value could be date-un-parseable String resulting in ParseException
3. Value could be parseable but in the wrong format. This is the worst case scenario since unlike the two before the parsing will be successful producing the wrong Date. (try Month/Day/Year format primarily used in the US).

Hopefully you understand now why we opted out of supporting certain properties in the initial implementation. 

So, we definitely need to give it more thought and I am curious to see if you have any. One thing to consider is  NiFi-wide StringDate representation which would have validation, documentation etc., 
to avoid issues described above.